client/components/grade-form.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+function InputFields(props) {
+
+  return (
+    <>
+      <div className="input-group mb-3">
+        <div className="input-group-prepend">
+          <span className="input-group-text"><i className="fas fa-user"></i></span>
+        </div>
+        <input type="text" className="form-control" onChange={props.onChange} placeholder="Name" name="name" aria-label="Name" />
+      </div>
+
+      <div className="input-group mb-3">
+        <div className="input-group-prepend">
+          <span className="input-group-text"><i className="fas fa-table"></i></span>
+        </div>
+        <input type="text" className="form-control" onChange={props.onChange} placeholder="Course" name="course" aria-label="Course" />
+      </div>
+
+      <div className="input-group mb-3">
+        <div className="input-group-prepend">
+          <span className="input-group-text"><i className="fas fa-graduation-cap"></i></span>
+        </div>
+        <input type="text" className="form-control" onChange={props.onChange} placeholder="Grade" name="grade" aria-label="Grade" />
+      </div>
+      <div className="btn-box justify-content-end d-flex">
+        <button className="btn-success pb-1" type="submit">Add</button>
+        <button className="btn-success ml-2 pb-1" onClick={props.resetForm} type="reset">Cancel</button>
+      </div>
+    </>
+  );
+
+}
+
+class GradeForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      course: '',
+      grade: null
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.resetForm = this.resetForm.bind(this);
+  }
+
+  resetForm() {
+    this.setState({
+      name: '',
+      grade: null,
+      course: ''
+    });
+  }
+
+  handleChange(event) {
+    this.setState({
+      name: event.target.value,
+      grade: event.target.value,
+      course: event.target.value
+    });
```

Comment:
When a change event occurs it should only update the state for the specific field. Use multiple change handlers if you need to. It's fine.

Reply:
Implemented the individual handlers for each input field to set state with